simple/lint.go

```diff
@@ -1639,3 +1640,47 @@ func (c *Checker) LintRedundantBreak(j *lint.Job) {
 		ast.Inspect(f, fn)
 	}
 }
+
+func (c *Checker) LintStringCopy(j *lint.Job) {
+	addEmptyStr := func(x, y ast.Expr) bool {
+		bl, ok := x.(*ast.BasicLit)
+		if !ok || bl.Kind != token.STRING || bl.Value != `""` {
+			return false
+		}
+		bt, ok := j.Program.Info.TypeOf(y).(*types.Basic)
```

Comment:
The type check seems counter-productive. Is `x + ""` not a redundant copy if `x` is of `type T string`? The fact that it's a string literal should be enough for soundness of the check. 

If I am mistaken, please add a test that demonstrates the importance of the check.